File: src/main/java/com/example/demo/service/TodoMockService.java
package com.example.demo.service;

import java.util.ArrayList;
import java.util.List;

import org.springframework.stereotype.Service;

import com.example.demo.entity.Todo;

@Service
public class TodoMockService implements ITodoService{

	private List<Todo> todos;

	
	public TodoMockService( ){
		
		todos = new ArrayList<Todo>();
		
		 todos.add(new Todo(1, "Setting Up The Angular5 front-end Project", "ts & html & css & service & model"));
		 todos.add(new Todo(2, "Setting Up The Spring boot back-end Project", "Java8 et service et controler"));
		 todos.add(new Todo(3, "Implementing the Add Task feature", "ADD"));	 
	}
	
	@Override
	public List<Todo> getTodos() {
		return todos;
	}

	@Override
	public void addTodo(Todo todo) {
		todos.add(todo);
		
	}

	@Override
	public void updateTodo(Todo todo) {
		todos.remove(todo);
		todos.add(todo);
	}

	@Override
	public void deleteTodo(int id) {
	Todo todo =new Todo();
	todo.setId(id);
	todos.remove(todo);	
		
	}

}

File: todo-app-ui/src/app/todo.mock.service.ts
import {Todo} from './shared/todo';
import {Injectable} from '@angular/core';


@Injectable()
export class TodoMockService {

 private TODOS:Todo[] = [];

constructor(){
    let t1: Todo = new Todo(1, 'Setting Up The Angular5 front-end Project', 'ts & html & css & service & model');
    let t2: Todo = new Todo(2, 'Setting Up The Spring boot back-end Project', 'java8 et service et controler');
    let t3: Todo = new Todo(3, 'Implementing the Add Task feature', 'CREATE');
    this.TODOS.push(t1);
    this.TODOS.push(t2);
    this.TODOS.push(t3);
}

public getTODOS() : Todo[] {
    return this.TODOS;
}
}